### ISSUE TITLE
Crazy CPU Usage

### ISSUE THREAD

**Issue by samczsun** [intellij-solidity/intellij-solidity#58]
I have a ~1000 line solidity file, and as soon as I open it IntelliJ CPU usage shoots up to 50-90% overall, and stays there. When I close the file CPU usage drops to acceptable (1-5%) amounts.

Is this just what happens with large files, or could there be a bug somewhere?

I'll be able to share the source code in a few days if that matters.

**Comment by SerCeMan** [comment]
Yes, an example of the problematic code might be really helpful.

**Comment by samczsun** [comment]
Here's a [gist](https://gist.github.com/samczsun/9db715b4c8aa74795814e89a8715f91c) of the code.

About my IntelliJ
```
IntelliJ IDEA 2017.3.5 (Ultimate Edition)
Build #IU-173.4674.33, built on March 5, 2018
Licensed to Sam Sun
Subscription is active until October 4, 2018
JRE: 1.8.0_152-release-1024-b15 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0
```

**Comment by 0v1se** [comment]
@samczsun could you try snapshot version of the plugin?
Just build it from the source and install in IDEA

**Comment by samczsun** [comment]
Installed 2.0.6-SNAPSHOT, still killing the CPU

**Comment by SerCeMan** [comment]
Hey, @samczsun!

It would be a great help if you could follow these instructions to create a CPU snapshot, https://intellij-support.jetbrains.com/hc/en-us/articles/207241235-Reporting-performance-problems and attach it to this ticket.

**Comment by samczsun** [comment]
Sounds good. Will do so when I have some more time this week

**Comment by samczsun** [comment]
So I took a CPU snapshot and noticed that it was getting hung up on `StubIndex` and `VirtualDirectoryImpl` which reminded me that I had a directory in this project with a couple thousand generated files. I marked those directories as excluded and solidity files no longer killed my computer.

Still seems like a bug that nothing else is choking on those files but at least my IDE is usable again.

Here's the snapshot if you're interested in taking a closer look: [IU-181.4445.78_samczsun_18.04.2018_18.32.51.zip](https://github.com/intellij-solidity/intellij-solidity/files/1926083/IU-181.4445.78_samczsun_18.04.2018_18.32.51.zip)
